src/components/card/event-card/index.js

```diff
@@ -1,11 +1,155 @@
+    onClickBookmark();
+  };
+
+  const renderVerticalCard = () => (
+    <VerticalCardContainer
+      direction={type}
+      hoverToLevel={3}
+      maxWidth={270}
+    >
+      <Link to={linkTo}><VerticalCover data-test="vertical-cover" src={cover} /></Link>
+      <VerticalContentContainer data-test="vertical-content">
+        <div>
+          <DateBookmarkContainer data-test="vertical-date-bookmark">
+            <ShowDate date={date} color="gray" fontSize="12" />
+            {
+              isBookmarked
+                ? <BookmarkIcon type="bookmark" size="lg" color="gray" onClick={handleClickBookmark} />
+                : <BookmarkIcon type="bookmark-border" size="lg" color="gray" onClick={handleClickBookmark} />
+            }
+          </DateBookmarkContainer>
+          <Link to={linkTo}><Title level={2} size="sm">{title}</Title></Link>
+          <VerticalPlacePriceContainer>
+            <div>
+              <PlacePriceIcon type="place" size="sm" color="gray" />
+              <Text color="gray" size="12">{place}</Text>
+            </div>
+            <div>
+              <PlacePriceIcon type="loyalty" size="sm" color="gray" />
+              <Text color="gray" size="12">{price}</Text>
+            </div>
+          </VerticalPlacePriceContainer>
+        </div>
+        {organization && (
+          <OrganizationLink to={`organizations/${organization.slug}`}>
+            <Avatar src={organization.logo} size="sm" />
+            <OrganizationName size="12">{organization.name}</OrganizationName>
+          </OrganizationLink>
+        )}
+      </VerticalContentContainer>
+    </VerticalCardContainer>
+  );
+
+  const renderHorizontalCard = () => (
+    <HorizontalCardContainer
+      direction={type}
+      hoverToLevel={3}
+      maxWidth={560}
+    >
+      <Link to={linkTo}><HorizontalCover data-test="horizontal-cover" src={cover} /></Link>
+      <HorizontalContentContainer data-test="horizontal-content">
+        <DateBookmarkContainer>
+          <ShowDate date={date} color="gray" fontSize="12" />
+          {
+            isBookmarked
+              ? <BookmarkIcon type="bookmark" size="lg" color="gray" onClick={handleClickBookmark} />
+              : <BookmarkIcon type="bookmark-border" size="lg" color="gray" onClick={handleClickBookmark} />
+          }
+        </DateBookmarkContainer>
+        <Link to={linkTo}><Title level={2} size="sm">{title}</Title></Link>
+        <HorizontalPlacePriceContainer>
+          <div>
+            <PlacePriceIcon type="place" size="sm" color="gray" />
+            <Text color="gray" size="12">{place}</Text>
+          </div>
+          <div>
+            <PlacePriceIcon type="loyalty" size="sm" color="gray" />
+            <Text color="gray" size="12">{price}</Text>
+          </div>
+        </HorizontalPlacePriceContainer>
+      </HorizontalContentContainer>
+    </HorizontalCardContainer>
+  );
+
+  const renderCard = () => {
+    switch (type) {
```

Comment:
it can't be one component and get a different style based on its type?

this implementation takes so many components (imported in this file line 9 to 24) with special styles for each of them.